catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/TimedSource.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.solr.impl;
+
+import ddf.catalog.data.ContentType;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.source.Source;
+import ddf.catalog.source.SourceMonitor;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TimedSource implements Source {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(TimedSource.class);
+
+  private final Source source;
+
+  public TimedSource(Source originalSource) {
+    source = originalSource;
+  }
+
+  @Override
+  public boolean isAvailable() {
+    return source.isAvailable();
+  }
+
+  @Override
+  public boolean isAvailable(SourceMonitor callback) {
+    return source.isAvailable(callback);
+  }
+
+  @Override
+  public SourceResponse query(QueryRequest request) throws UnsupportedQueryException {
+    long startTime = System.currentTimeMillis();
+    SourceResponse result = source.query(request);
+    long endTime = System.currentTimeMillis();
+
+    int elapsedTime = Math.toIntExact(endTime - startTime);
+
+    result.getProperties().put("metrics.source.elapsed." + source.getId(), elapsedTime);
```

Comment:
✏️ Using this magic string here with the concatenation is a bit brittle. It would be nice if we had a centralized place where the prefix lived, ideally with three methods like:
1. `createSourceElapsedPropKey(String): String`
2. `isSourceElapsedPropKey(String): Boolean`
3. `getSourceFromKey(String): String`

The first would only be needed here; the latter two would be used in `SourceMetricsImpl`.